components/Grid/GridColumn.vue

```diff
@@ -63,8 +109,312 @@ export default {
+                clientX,
+                columnXPos,
+                columnWidth,
+            );
+            const fixedIndex = this.getColumnFixedIndex();
+
+            if ((this.index === this.draggedElIndex && this.ghostIndex !== -1)
+                || (isBefore && this.ghostIndex === fixedIndex - 1)
+                || (!isBefore && this.ghostIndex === fixedIndex + 1)
+                || (this.isExtenderColumn && !isBefore)
+                || this.isPinnedLeft
+                || this.isPinnedRight) {
+                return false;
+            }
+
+            if (this.ghostIndex === -1) {
+                this.insertColumnOnDragOver(isBefore, columnXPos, columnWidth);
+            } else {
+                this.updateColumnsTransition(isBefore);
+                this.updateGhostIndex(isBefore);
+            }
+
+            return true;
+        },
+        onDragLeave(event) {
+            const { pageX, pageY } = event;
+
+            if (pageX === 0 && pageY === 0) return false;
+
+            const elementBelowMouse = document.elementFromPoint(pageX, pageY);
+            const grid = document.documentElement.querySelector('.grid');
+            const isOutOfBounds = isMouseOutOfBoundsElement(grid, pageX, pageY);
+            const ghostColumnExists = this.draggedElIndex >= 0;
+            const isTrashBelowMouse = elementBelowMouse && elementBelowMouse.className === 'trash-can';
+
+            if ((isOutOfBounds || isTrashBelowMouse) && ghostColumnExists) {
+                for (let i = 0; i < grid.children.length; i += 1) {
+                    grid.children[i].style.transform = 'translateX(0)';
+                }
+                this.removeColumnAtIndex({ index: this.draggedElIndex });
+                this.setGhostIndex();
+                this.setDraggedElIndex();
+                this.setGhostElTransform();
+            }
+
+            return true;
+        },
+        initResizeDrag(event) {
+            this.isResizing = true;
+            this.initMousePosition(event);
+            this.initElementWidth();
+            this.initElementStyleForResizeState();
+            this.addEventListenersForResizeState();
+        },
+        doResizeDrag(event) {
+            const { clientX } = event;
+            const width = this.getElementWidthBasedOnMouseXPosition(clientX);
+
+            if (width > this.column.minWidth) {
+                this.columnWidth = width;
+                this.updateElementWidth(width);
+            }
+        },
+        stopResizeDrag() {
+            this.updateColumnWidthAtIndex({
+                index: this.index, width: this.columnWidth,
+            });
+            this.removeEventListenersForResizeState();
+            this.isResizing = false;
+        },
+        initMousePosition({ clientX }) {
+            this.startX = clientX;
+        },
+        initElementWidth() {
+            const {
+                width: elementWidth,
+            } = this.$el.getBoundingClientRect();
```

Comment:
you use `this.$el.getBoundingClientRect();` a couple of times. Maybe you can put it in the method.

Reply:
Why would I ? It is already method call